### ISSUE TITLE
Error while upgrading iPhone 12

### ISSUE THREAD

**Issue by Exormeter** [marijuanARM/futurerestore#57]
Hello, I encountered an error on the latest futureRestore trying to upgrade an iPhone 12 on iOS 14.2.1 to 15.1.

```
Personalizing IMG4 component iBoot...
Personalizing IMG4 component RestoreSEP...
Personalizing IMG4 component SEP...
common.c:supressed printing 27854450 bytes plist...
Sending NORData now...
Done sending NORData
Cleaning up...
[exception]:
what=ERROR: Unable to restore device

code=82182161
line=1254
file=futurerestore.cpp
commit count=266:
commit sha  =06137f3ba489b5ad89fa0f2a8930305e6ce7ee92:
Done: restoring failed!
```

I tried it serveral times, but it always gets stuck here. I am able to kick it out of recovery though. Is there any fix?
Thanks

**Comment by Cryptiiiic** [comment]
No fix, move on with your life

**Comment by Cryptiiiic** [comment]
This is a joke, I haven't looked into fix yet

**Comment by Exormeter** [comment]
So it is a known bug?

**Comment by HydrationMan** [comment]
Yes

**Comment by JiNama** [comment]
I'm in the same boat, A14 w/ iOS 14.2.1 trying to upgrade to 15.1. Wish I'd learned about blobs while 14.3 was still being signed lol.

**Comment by camelfert** [comment]
Same issue, 12 Pro Max on 14.4.2 to 15.1.1 errors out about NORData and says Unable to restore device. Some other people in #futurerestore-help on the r/jailbreak Discord are experiencing the same.

**Comment by alicans** [comment]
```
Sending NORData now...
Done sending NORData
ERROR: Unable to receive message from FDR 0x60000305c4c0 (-2). 0/2 bytes
ERROR: Could not read data (-256). Aborting.
Cleaning up...
[exception]:
what=ERROR: Unable to restore device

code=82182161
line=1254
file=futurerestore.cpp
commit count=266:
commit sha  =06137f3ba489b5ad89fa0f2a8930305e6ce7ee92:
Done: restoring failed!
```

**Comment by luochen28** [comment]
yes，me too![image](https://user-images.githubusercontent.com/98925125/156287751-9a41b6c4-7dcc-41e5-a26f-35ea5c3703c2.png)

**Comment by ThomasAarts** [comment]
I have the same issue jailbroken on 14.1 with unc0ver and have specific 15.2Beta1 blobs trying to FutureRestore also heard that 15.2Beta1 blobs seem to have baseband compatability issues! Really hoping and praying that this gets figured out before 15.3.1 gets unsigned...

**Comment by f0enix** [comment]
@Exormeter @JiNama @Fr0styCam have you tried to futurerestore to 15.0 or 15.0.1 to see if it goes through? Maybe the issue is only on 15.1.x?

**Comment by alicans** [comment]
i tried on 15.0 and 15.0.1 but no luck..
iphone 12 14.2.1

**Comment by f0enix** [comment]
> i tried on 15.0 and 15.0.1 but no luck..

Can you share the full verbose log?

**Comment by alicans** [comment]
> > i tried on 15.0 and 15.0.1 but no luck..
> 
> Can you share the full verbose log?

```
common.c:supressed printing 27870714 bytes plist...
Sending NORData now...
Done sending NORData
idevice_event_cb: device 0010393e1af8001e (udid: ..) disconnected
ERROR: Unable to receive message from FDR 0x600000a70380 (-2). 0/2 bytes
FDR 0x600000a70380 terminating...
ERROR: Could not read data (-256). Aborting.
Cleaning up...
[exception]:
what=ERROR: Unable to restore device

code=82182161
line=1254
file=futurerestore.cpp
commit count=266:
commit sha  =06137f3ba489b5ad89fa0f2a8930305e6ce7ee92:
Done: restoring failed!
```

full log is here: https://www.dropbox.com/s/ub9hosa0fdqsrnv/log.rtf?dl=0

**Comment by f0enix** [comment]
> > > i tried on 15.0 and 15.0.1 but no luck..
> 
> > 
> 
> > Can you share the full verbose log?
> 
> 
> 
> ```
> 
> common.c:supressed printing 27870714 bytes plist...
> 
> Sending NORData now...
> 
> Done sending NORData
> 
> idevice_event_cb: device 0010393e1af8001e (udid: ..) disconnected
> 
> ERROR: Unable to receive message from FDR 0x600000a70380 (-2). 0/2 bytes
> 
> FDR 0x600000a70380 terminating...
> 
> ERROR: Could not read data (-256). Aborting.
> 
> Cleaning up...
> 
> [exception]:
> 
> what=ERROR: Unable to restore device
> 
> 
> 
> code=82182161
> 
> line=1254
> 
> file=futurerestore.cpp
> 
> commit count=266:
> 
> commit sha  =06137f3ba489b5ad89fa0f2a8930305e6ce7ee92:
> 
> Done: restoring failed!
> 
> ```
> 
> 
> 
> full log is here: https://www.dropbox.com/s/ub9hosa0fdqsrnv/log.rtf?dl=0

Can you try going to 15.1 but using custom sep and baseband of 15.0.2? The sep flashing is failing when using latest sep.

**Comment by alicans** [comment]
> > > > i tried on 15.0 and 15.0.1 but no luck..
> > 
> > 
> > > 
> > 
> > 
> > > Can you share the full verbose log?
> > 
> > 
> > ```
> > 
> > common.c:supressed printing 27870714 bytes plist...
> > 
> > Sending NORData now...
> > 
> > Done sending NORData
> > 
> > idevice_event_cb: device 0010393e1af8001e (udid: ..) disconnected
> > 
> > ERROR: Unable to receive message from FDR 0x600000a70380 (-2). 0/2 bytes
> > 
> > FDR 0x600000a70380 terminating...
> > 
> > ERROR: Could not read data (-256). Aborting.
> > 
> > Cleaning up...
> > 
> > [exception]:
> > 
> > what=ERROR: Unable to restore device
> > 
> > 
> > 
> > code=82182161
> > 
> > line=1254
> > 
> > file=futurerestore.cpp
> > 
> > commit count=266:
> > 
> > commit sha  =06137f3ba489b5ad89fa0f2a8930305e6ce7ee92:
> > 
> > Done: restoring failed!
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > full log is here: https://www.dropbox.com/s/ub9hosa0fdqsrnv/log.rtf?dl=0
> 
> Can you try going to 15.1 but using custom sep and baseband of 15.0.2? The sep flashing is failing when using latest sep.

is 15.0.2 sep compatible with 15.1? or trying 15.3.1 custom sep and baseband?

**Comment by camelfert** [comment]
> > > > i tried on 15.0 and 15.0.1 but no luck..
> > 
> > 
> > > 
> > 
> > 
> > > Can you share the full verbose log?
> > 
> > 
> > ```
> > 
> > common.c:supressed printing 27870714 bytes plist...
> > 
> > Sending NORData now...
> > 
> > Done sending NORData
> > 
> > idevice_event_cb: device 0010393e1af8001e (udid: ..) disconnected
> > 
> > ERROR: Unable to receive message from FDR 0x600000a70380 (-2). 0/2 bytes
> > 
> > FDR 0x600000a70380 terminating...
> > 
> > ERROR: Could not read data (-256). Aborting.
> > 
> > Cleaning up...
> > 
> > [exception]:
> > 
> > what=ERROR: Unable to restore device
> > 
> > 
> > 
> > code=82182161
> > 
> > line=1254
> > 
> > file=futurerestore.cpp
> > 
> > commit count=266:
> > 
> > commit sha  =06137f3ba489b5ad89fa0f2a8930305e6ce7ee92:
> > 
> > Done: restoring failed!
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > full log is here: https://www.dropbox.com/s/ub9hosa0fdqsrnv/log.rtf?dl=0
> 
> Can you try going to 15.1 but using custom sep and baseband of 15.0.2? The sep flashing is failing when using latest sep.

I just tried that and it didn't work either, it errors at the same spot again. I also tried with 15.3.1's SEP and it failed in the same spot too.

**Comment by idoktorz** [comment]
Here is my full log if might help:
https://pastebin.com/yLzg06UH

**Comment by Cryptiiiic** [comment]
[futurerestore-macOS-local.zip](https://github.com/m1stadev/futurerestore/files/8173673/futurerestore-macOS-local.zip)
Can you guys test this build on a14?
@Exormeter @JiNama @Fr0styCam @alicans @luochen28 @ThomasAarts @idoktorz

**Comment by Cryptiiiic** [comment]
This is confirmed fixing NORData ^

**Comment by Cryptiiiic** [comment]
Fixed in 6406f51
